build.sc

```diff
@@ -102,25 +102,31 @@ trait ItestCross extends Cross.Module[String] with MillIntegrationTestModule {
 
   override def testInvocations: T[Seq[(PathRef, Seq[TestInvocation.Targets])]] =
     T {
+      val env = if(millTestVersion() >= "0.12")
+        Map(
+          "COURSIER_REPOSITORIES" -> s"central sonatype:releases ivy:file://${T.dest.toString.replaceFirst("testInvocations", "test")}/ivyRepo/local/[organisation]/[module]/[revision]/[type]s/[artifact].[ext]"
+        )
+      else
+        Map.empty[String, String]
       Seq(
-        PathRef(testBase / "minimal") -> Seq(
-          TestInvocation.Targets(Seq("checkManifest"), noServer = true)
-        ),
-        PathRef(testBase / "directRelationship") -> Seq(
-          TestInvocation.Targets(Seq("verify"), noServer = true)
-        ),
-        PathRef(testBase / "eviction") -> Seq(
-          TestInvocation.Targets(Seq("verify"), noServer = true)
-        ),
-        PathRef(testBase / "range") -> Seq(
-          TestInvocation.Targets(Seq("verify"), noServer = true)
-        ),
-        PathRef(testBase / "reconciledRange") -> Seq(
-          TestInvocation.Targets(Seq("verify"), noServer = true)
-        ),
-        PathRef(testBase / "cyclical") -> Seq(
-          TestInvocation.Targets(Seq("checkManifest"), noServer = true)
+        "minimal" -> "checkManifest",
+        "directRelationship" -> "verify",
+        "eviction" -> "verify",
+        "range" -> "verify",
+        "reconciledRange" -> "verify",
+        "cyclical" -> "checkManifest"
+      ).filter {
+          // The reconciledRange test doesn't work under 0.12 atm
```

Comment:
So this was due to what I believe is a bug in coursier. I created an issue for it in https://github.com/coursier/coursier/issues/2682 but never got a response about what is actually happening. Maybe this is fixed with the new changes in coursier, but I'd need to dig deeper to better understand if that's the case. Right now you can go ahead and just ignore this and I'll try to look deeper at what's happening.

Reply:
I dug a bit deeper and created https://github.com/com-lihaoyi/mill/issues/4165.